File: adapters/ios-keyboard/swift/KhmerIMEKeyboard/Views/GlassColorSpec.swift
import UIKit

enum GlassColorSpec {

    // Glass surface for individual keys.
    static func backgroundColor(isDark: Bool) -> UIColor {
        isDark
            ? UIColor(red: 90/255, green: 90/255, blue: 100/255, alpha: 210/255)
            : UIColor(red: 1, green: 1, blue: 1, alpha: 230/255)
    }

    // Background of the whole keyboard body (darker than the keys).
    static func keyboardBackground(isDark: Bool) -> UIColor {
        isDark
            ? UIColor(white: 0.15, alpha: 0.95)
            : UIColor(red: 209/255, green: 213/255, blue: 219/255, alpha: 1)
    }

    static func borderColor(isDark: Bool) -> UIColor {
        isDark
            ? UIColor(white: 1, alpha: 60/255)
            : UIColor(white: 0, alpha: 40/255)
    }

    static func toggleActiveBackground(isDark: Bool) -> UIColor {
        isDark
            ? UIColor(red: 240/255, green: 240/255, blue: 245/255, alpha: 245/255)
            : UIColor(red: 1, green: 1, blue: 1, alpha: 245/255)
    }

    static func toggleActiveTextColor() -> UIColor {
        UIColor(red: 20/255, green: 20/255, blue: 24/255, alpha: 1)
    }

    static func selectedCandidateBackground(isDark: Bool) -> UIColor {
        isDark
            ? UIColor(red: 110/255, green: 110/255, blue: 124/255, alpha: 235/255)
            : UIColor(red: 210/255, green: 215/255, blue: 225/255, alpha: 245/255)
    }

    static func candidateBorderWidth() -> CGFloat { 1.5 }

    static func blurRadius(scale: CGFloat) -> CGFloat { 12 * scale }
}

File: adapters/android-ime/app/src/main/java/com/khmerime/views/GlassColorSpec.kt
package com.khmerime

object GlassColorSpec {
    fun backgroundColor(isDark: Boolean): Int =
        if (isDark) argb(210, 90, 90, 100) else argb(230, 255, 255, 255)

    fun borderColor(isDark: Boolean): Int =
        if (isDark) argb(60, 255, 255, 255) else argb(40, 0, 0, 0)

    fun toggleActiveBackground(isDark: Boolean): Int =
        if (isDark) argb(245, 240, 240, 245) else argb(245, 255, 255, 255)

    fun toggleActiveTextColor(): Int = argb(255, 20, 20, 24)

    fun selectedCandidateBackground(isDark: Boolean): Int =
        if (isDark) argb(235, 110, 110, 124) else argb(245, 210, 215, 225)

    fun candidateBorderWidth(): Float = 1.5f

    fun blurRadiusPx(density: Float): Float = 12f * density

    private fun argb(a: Int, r: Int, g: Int, b: Int): Int =
        (a and 0xFF shl 24) or (r and 0xFF shl 16) or (g and 0xFF shl 8) or (b and 0xFF)
}
